### PR TITLE
Remove petset documentation

### PR DESCRIPTION
**Pull request by crimsonfaith91**
As 1.8 will roll out soon, we will not support petset anymore. Hence, all relevant documentations can be removed.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5395)

### CODE REVIEW

File: skip_toc_check.txt [hunk 1]
```diff
@@ -96,8 +96,6 @@ docs/user-guide/namespaces.md
 docs/user-guide/networkpolicies.md
 docs/user-guide/node-selection/index.md
 docs/user-guide/persistent-volumes/index.md
-docs/user-guide/petset.md
```
Comment (crimsonfaith91) [inline skip_toc_check.txt]: Not sure whether we should change this file or not.

File: _redirects [hunk 2]
```diff
@@ -41,7 +41,6 @@
 /docs/api														/docs/concepts/overview/kubernetes-api		301
 
 /docs/concepts/abstractions/controllers/garbage-collection		/docs/concepts/workloads/controllers/garbage-collection		301
-/docs/concepts/abstractions/controllers/petsets					/docs/concepts/workloads/controllers/petset		301
```
Comment (crimsonfaith91) [inline _redirects]: Based on [netlify website](https://www.netlify.com/docs/redirects/), i think we should remove petset items from the redirects.

File: _data/concepts.yml [hunk 3]
```diff
@@ -47,7 +47,6 @@ toc:
     - docs/concepts/workloads/controllers/replicationcontroller.md
     - docs/concepts/workloads/controllers/deployment.md
     - docs/concepts/workloads/controllers/statefulset.md
-    - docs/concepts/workloads/controllers/petset.md
```
Comment (crimsonfaith91) [inline _data/concepts.yml]: Not sure a change to `_data` is needed or not.

File: docs/concepts/workloads/pods/pod.md [hunk 4]
```diff
@@ -150,7 +150,7 @@ Pod is exposed as a primitive in order to facilitate:
 * clean composition of Kubelet-level functionality with cluster-level functionality &mdash; Kubelet is effectively the "pod controller"
 * high-availability applications, which will expect pods to be replaced in advance of their termination and certainly in advance of deletion, such as in the case of planned evictions, image prefetching, or live pod migration [#3949](http://issue.k8s.io/3949)
 
-There is new first-class support for stateful pods with the [StatefulSet](/docs/concepts/abstractions/controllers/statefulsets/) controller (currently in beta). The feature was alpha in 1.4 and was called [PetSet](/docs/concepts/workloads/controllers/petset/). For prior versions of Kubernetes, best practice for having stateful pods is to create a replication controller with `replicas` equal to `1` and a corresponding service, see [this MySQL deployment example](/docs/tutorials/stateful-application/run-stateful-application/).
```
Comment (crimsonfaith91) [inline docs/concepts/workloads/pods/pod.md]: We have to change `/docs/concepts/abstractions/controllers/statefulsets/` to `/docs/concepts/workloads/controllers/statefulset.md`.

The reason is most files of `/docs/concepts/abstractions/controllers/` have been removed: [link](https://github.com/kubernetes/kubernetes.github.io/tree/master/docs/concepts/abstractions/controllers)

File: _data/concepts.yml (hunk 3 above)
Comment (kow3ns) [reply _data/concepts.yml]: should update when you build the documentations

File: _redirects (hunk 2 above)
Comment (kow3ns) [reply _redirects]: +1

File: _data/concepts.yml (hunk 3 above)
Comment (crimsonfaith91) [reply _data/concepts.yml]: thanks for verifying that a change to `_data` is needed. :)

**Comment by crimsonfaith91** [comment]
@steveperry-53 Do you mind to provide `Docs LGTM` and `Tech Review LGTM` for this PR if everything looks good to you? Thanks!

**Comment by zhangxiaoyu-zidif** [comment]
our kubernetes.io contains v1.5, 1.6, 1.7's contents.  we can switch version by  website's version choices. I am not sure if it affects v1.5, v1.6's usage.

**Comment by crimsonfaith91** [comment]
@zhangxiaoyu-zidif It will not - we will not support petset anymore in 1.8 (it is completely retired).

File: _redirects (hunk 2 above)
Comment (steveperry-53) [reply _redirects]: I don't think this is correct. We don't want to remove any URLs from the left column of _redirects. We want to keep all the existing URLs in the left column, but change the target URL in the right column.

Also, for each topic that is being removed from the doc set, add the URL of the removed topic to the left column of _redirects, and put an appropriate target URL in the right column.

**Comment by steveperry-53** [comment]
@crimsonfaith91, One comment for you in _redirects.

File: _redirects [hunk 5]
```diff
@@ -41,6 +41,7 @@
 /docs/api														/docs/concepts/overview/kubernetes-api		301
 
 /docs/concepts/abstractions/controllers/garbage-collection		/docs/concepts/workloads/controllers/garbage-collection		301
+/docs/concepts/abstractions/controllers/petsets							301
```
Comment (crimsonfaith91) [inline _redirects]: @steveperry-53 The reason I initially removed all `petset` links as `petset` is officially retired for 1.8 when it rolls out. Based on your feedback, I retained all links at left side. As `petset` is retired, there should not be any link to its documentation, so i still removed all `petset` links at right side. Is it correct?

Should i also remove `301` if all the `petset` links are removed at right side?

**Comment by steveperry-53** [comment]
@crimsonfaith91 The redirects should look like this:

/docs/concepts/abstractions/controllers/petsets   /docs/concepts/workloads/controllers/statefulset 301

/docs/user-guide/petset   /docs/concepts/workloads/controllers/statefulset 301

/docs/user-guide/petset/bootstrapping   /docs/concepts/workloads/controllers/statefulset 301

/docs/abstractions/controllers/petset   /docs/concepts/workloads/controllers/statefulset 301

/docs/concepts/workloads/controllers/petsets   /docs/concepts/workloads/controllers/statefulset 301

If you'd like me to make the changes to _redirects, you can check the **Allow edits by maintainers** box in the right column.

**Comment by crimsonfaith91** [comment]
@steveperry-53 Noted! I will send out a commit soon. Thanks!

**Comment by crimsonfaith91** [comment]
@steveperry-53 If everything looks good to you, i will squash.

**Comment by steveperry-53** [comment]
@crimsonfaith91 lgtm, Go ahead and squash.

**Comment by crimsonfaith91** [comment]
@steveperry-53 squashed the commit.

**Pull request merged**